### ISSUE TITLE
Fix S101: Rule should not report an issue on auto-generated members

### ISSUE THREAD

**Issue by fmallet** [SonarSource/sonar-csharp#318]
The rule should not report on the class (or all parts of a partial class) if we detect the class was auto-generated.

Example:
file1:
```
///<auto-generated>
partial class SOME_CLASS
{
}
```

file2:
```
partial class SOME_CLASS // Noncompliant, although it can't be changed due to the other part.
{
}
```
